components/admin/requests/processing/TasksCard.tsx

```diff
@@ -90,20 +93,50 @@ export default function ProcessingTasksCard({ applicationId }: ProcessingTasksCa
     refetch();
   };
 
-  // TODO: Hook up to S3 upload
   const [uploadDocuments] =
     useMutation<UploadDocumentsResponse, UploadDocumentsRequest>(UPLOAD_DOCUMENTS_MUTATION);
-  const handleUploadDocuments = async (documentsUrl: string) => {
-    await uploadDocuments({ variables: { input: { applicationId, documentsUrl } } });
-    refetch();
-  };
 
   const [mailOut] = useMutation<MailOutResponse, MailOutRequest>(MAIL_OUT_APP_MUTATION);
   const handleMailOut = async (appMailed: boolean) => {
     await mailOut({ variables: { input: { applicationId, appMailed } } });
     refetch();
   };
 
+  const toast = useToast();
+  const [applicationDocuments, setApplicationDocuments] = useState<File | null>(null);
+
+  const handleSubmitDocuments = async (applicationDoc: File) => {
+    setApplicationDocuments(applicationDoc);
+    let documentS3ObjectKey = '';
+    if (applicationDoc) {
+      try {
+        const { key } = await clientUploadToS3(applicationDoc, 'rcd/application-documents');
+        documentS3ObjectKey = key;
+      } catch (err) {
+        toast({
+          status: 'error',
+          description: `Failed to upload documents: ${err}`,
+          isClosable: true,
+        });
+        setApplicationDocuments(null);
+        return;
+      }
+    }
```

Comment:
couple things here. Can we:
- remove the state var and the if condition (file should always be present)
- rename the key variable when destructuring instead of creating a new variable

Could look something like
```ts
    try {
      const { key: documentS3ObjectKey } = await clientUploadToS3(applicationDoc, 'rcd/application-documents');
    } catch (err) {
      toast({
        status: 'error',
        description: `Failed to upload documents: ${err}`,
        isClosable: true,
      });
      return;
    }
```

Reply:
Here documentS3ObjectKey becomes out of scope for the uploadDocuments mutation that's outside the try catch. Should the mutation be moved within the try catch block as well ?

Reply:
ah i see good point. Let's just keep it the same as before then where `documentS3ObjectKey` is declared outside the try catch then assigned inside